### PR TITLE
Remove normalizeProjections to prevent field/collection name collision

### PR DESCRIPTION
**Pull request by edmundnoble**
Turns out, `normalizeProjections` was causing this issue and isn't necessary anymore.

#qz-3649 Done

### CODE REVIEW

**Comment by alissapajer** [comment]
@edmundnoble can we add a new dataset with a field name the same as the dataset name and then project out the field, to show that it works?

Tests that return `[]` aren't bad to have, but we should strive to have their positive equivalent as well.

**Comment by alissapajer** [comment]
`./regression/fieldtablecollision.test` is failing for mongo.

**Comment by edmundnoble** [comment]
@alissapajer This test should be better, and works on Mongo locally.

File: it/src/main/resources/tests/regression/fieldtablecollision.test [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "[reg] field/table name collision",
+  "data": "fieldtablecollision.data",
+  "backends": {
+  },
+  "query": "SELECT fieldtablecollision.fieldtablecollision FROM fieldtablecollision",
```
Comment (alissapajer) [inline it/src/main/resources/tests/regression/fieldtablecollision.test]: :smile:

**Pull request merged**